src/workflows/publish.sh

```diff
@@ -0,0 +1,73 @@
+#!/bin/bash
+
+set -o nounset
+set -o pipefail
+set -o errexit
+
+fail_if_unset () {
+  local var_name="$1"
+  if [[ -z "${!var_name:-""}" ]]; then
+    printf '%s not set\n' "$var_name" >&2
+    exit 1
+  fi
+}
+
+fail_if_unset GITLAB_TRIGGER_TOKEN
+fail_if_unset BUCKET_NAME
+fail_if_unset PUBLISH_SERVER
+fail_if_unset AWS_ACCESS_KEY_ID
+fail_if_unset AWS_SECRET_ACCESS_KEY
+fail_if_unset AWS_REGION
+
+git_username="GitHub Actions"
+git_useremail="GitHub-Actions@GP-v2-contracts"
```

Comment:
Tiny nit, but maybe this should be identical to the email we used when creating the PR with the updated deployment files:
![image](https://user-images.githubusercontent.com/4210206/110459365-302f7680-80cd-11eb-8b37-e3ebb044bf9f.png)